nrf-softdevice-macro/src/lib.rs

```diff
@@ -48,12 +48,15 @@ pub fn gatt_server(_args: TokenStream, item: TokenStream) -> TokenStream {
     let struct_fields = match &mut struc.fields {
         syn::Fields::Named(n) => n,
         _ => {
-            struc
+            let s = struc
                 .ident
                 .span()
-                .unwrap()
-                .error("gatt_server structs must have named fields, not tuples.")
+                .unwrap();
+
+            #[cfg(feature = "nightly")]
```

Comment:
This means errors won't be reported at all without nightly, the macro will silently allow wrong things through.

Perhaps copy this, that allows having the same functionality on stable https://github.com/embassy-rs/embassy/blob/master/embassy-macros/src/util/ctxt.rs

example on how to use it: https://github.com/embassy-rs/embassy/blob/master/embassy-macros/src/macros/main.rs#L71-L84

Reply:
Oh, i expected it to just blow up without any proper error reporting. On it!